input-output-hk/lace
# [LW-7681] lace-blockchain-services: HTTP API to control the network switch and view available services

# Checklist

- [ ] JIRA - https://input-output.atlassian.net/browse/LW-7681

---

## Proposed solution

Define a few API endpoints to control the network switch, and service discovery.

## Testing

The following should work the same **on all 3 platforms** (4, if you count macOS on Apple Silicon separately).

1. From the system tray menu, select "**OpenAPI Specification v3.0**", and click around.

    1. Submit a `/v1/switch-network` from the Swagger UI ("Try it out" → "Execute"). It should switch the network (same effect as selecting that from the system tray UI). Check logs, the tray menu, and the WebSocket output.

    1. Check if the `/v1/info` endpoint is responding with reasonable data (compared to what's displayed in the tray menu).

1. From the system tray menu, select "**Dashboard**", and click around.

    1. In there, watch what's being written to the WebSocket (`/v1/websocket`) when you switch networks, watch sync progress etc.

    1. Try switching networks by pressing the three NetworkMagic buttons on the left side (see the recording below).

## Screenshots

<table>
<tr>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/5d2f75a4-8d4b-4846-9de6-86ad47fc63fb">
</td>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/31373e5b-c5b1-408d-9224-418ad3d798f6">
</td>
</tr>
<tr>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/dc091532-10c8-4de5-9feb-19db6d141227">
</td>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/7cf34feb-503f-4061-a5a0-0c1b6f85d4a7">
</td>
</tr>
<tr>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/1ef4db64-705b-41d4-bfa0-e8e87100ce00">
</td>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/7c326cb2-110b-4271-b3d3-d5fec039bf37">
</td>
</tr>
<tr>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/26b6173e-566f-4b18-a564-73688cf24b58">
</td>
<td>
<img src="https://github.com/input-output-hk/lace/assets/4366292/3873e51d-3acd-4590-bbe0-45d9ec238083">
</td>
</tr>
</table>



---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/1297/5750546822/index.html) for [9c26fbe4](https://github.com/input-output-hk/lace/pull/319/commits/9c26fbe445a5b2dcb68a24b30fa4ac0f368bf897)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 33     | 3      | 0       | 0     | 36    | ❌     |
